ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryIdConverterFactory.java

```diff
@@ -215,9 +216,11 @@ private CompletionStage<String> convertId(String input, RestRequest restRequest,
           String blobId = RestUtils.stripSlashAndExtensionFromId(input);
           long expirationTimeMs = Utils.addSecondsToEpochTime(blobProperties.getCreationTimeInMs(),
               blobProperties.getTimeToLiveInSeconds());
+          byte[] digestBytes = restRequest.getDigest();
```

Comment:
can we confirm if digest is already calculated and available here?  there could be 'chunked' encoding requests where md5 is available after all data is processed. so my understanding is the digest used here is not correct.

can you also add a test to 
1) generate some content - c1
2) calculate md5 of c1 -  d1
3) send put request with c1
4) verify digest is stored as d1 in mysql database